src/test/scala/com/vngrs/etl/PipelineSpec.scala

```diff
@@ -0,0 +1,199 @@
+package com.vngrs.etl
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.{SparkContext, SparkException}
+
+/**
+  * Pipeline Specifications
+  */
+// Following wart does not work with scalatest's intercept functionality
+@SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+class PipelineSpec extends BaseSparkSpec {
+  "A Pipeline" should "accept an Empty RDD" in {
+    val rdd = emptyRDD[Int]
+
+    val pipeline = new Pipeline(rdd)
+
+    pipeline.rdd should equal (rdd)
```

Comment:
Since we can not give full ability of the `RDD`s to `Pipeline` and we should make it happen somehow, this is how we do it :). I believe it is required to keep project updated when a new feature is added to `RDD` so that users can still use those abilities. One could argue that new feature may not be suitable with the `Pipeline`. Since `RDD`s are immutable. it will not affect `Pipeline` internal state. So, I believe it is okay.